build/dart/rules.gni

```
@@ -390,3 +390,117 @@ template("application_snapshot") {
     }
   }
 }
+
+# Generates an executable from Dart code using `dart compile exe`.
+#
+# This rule is primarily intended to be used to compile test targets for use
+# by engine developers to build and run test code written in Dart in the engine
+# repo, in particular via the `et` tool. It is NOT intended for use in
+# generating production-ready user-facing binaries.
+#
+# See: //flutter/tools/engine_tool for more information on `et`.
+#
+# Arguments
+#  main_dart (required):
+#    The Dart entrypoint file.
+#
+#  package_config (optional):
+#    The packages.json file for the Dart package (if any). Required when
+#    building libraries that are part of packages with Dart dependencies.
+#    Not needed when building standalone Dart code that is not part of a
+#    package.
+#
+#  output (optional):
+#    The output executable name. By default, the target name. On Windows, an
+#    'exe' is automatically appended.
+#
+#  deps (optional):
+#    Additional dependencies. Dependencies on the frontend server and
+#    Flutter's platform.dill are included by default. This rule creates and
+#    uses a depfile, so listing all Dart sources is not necessary.
+template("dart_executable") {
+  assert(defined(invoker.main_dart), "Must specify 'main_dart'")
+
+  main_dart = invoker.main_dart
+
+  name = target_name
+
+  extra_deps = []
+  if (defined(invoker.deps)) {
+    extra_deps += invoker.deps
+  }
+  extra_inputs = [ main_dart ]
+  if (defined(invoker.inputs)) {
+    extra_inputs += invoker.inputs
+  }
+  if (defined(invoker.package_config)) {
+    package_config = rebase_path(invoker.package_config)
```

I think all these paths can be relative to the root build dir:
```suggestion
    package_config = rebase_path(invoker.package_config, root_build_dir)
```

---

Done.